sycl/test/Unit/lit.cfg.py

```diff
@@ -9,6 +9,8 @@
 import lit.formats
 import lit.util
 
+from lit.llvm import llvm_config
```

Comment:
Note for reviewers: I wanted to disable all defaults for `SYCL_DEVICELIB_NO_FALLBACK` option, but it could be done in a simpler way. If necessary, I can outline `llvm_config` addition into a separate PR, refactoring other places where we propagate system environment into test environment all together.

Reply:
Am I understanding correctly that it is intentionally removing the default `1` for `SYCL_DEVICE_LIB_NO_FALLBACK`? If so, can we still call it NFC?

Reply:
I always perceived `NFC` as no functional changes in the product, but perhaps you are right. From a developer point of view this is a functional change to the testing infrastructure

Reply:
The latter was my understanding, but I am not sure. We could consider using tags for testing PRs to differentiate.